README.md
# gatsby-remark-images-insert-wrapper-attributes

![Coverage](https://img.shields.io/badge/coverage-100%25-brightgreen) ![Circle CI](https://img.shields.io/badge/build-passing-brightgreen) ![License](https://img.shields.io/badge/license-MIT-blue)

Insert AST properties into the wrapper generated by Gatsby Remark Images.

### Align And Scale Markdown Images
This plugin is used in conjunction with the [gatsby-remark-extract-image-attributes](https://github.com/newsproutsmedia/gatsby-remark-extract-image-attributes) plugin to extract image attributes and then apply them to the wrapper generated by [gatsby-remark-images](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-images) after image processing.

Using the two plugins together makes it simple to align and size images in your Markdown files while still taking advantage of Gatsby's excellent image optimization plugin, [gatsby-remark-images](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-images).

## Getting Started

### Compatibility
This plugin has been tested, and is compatible, with the following (related) dependencies: 

- gatsby: ^3.3.1
- gatsby-transformer-remark: ^3.0.0
- gatsby-transformer-sharp: ^3.0.0
- gatsby-remark-images: ^4.0.0

### Prerequisites
- [gatsby-transformer-remark](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-remark)
- Node.js v10 or newer 

### Installation
```
npm install gatsby-remark-images-insert-wrapper-attributes
```
## Configuration
The gatsby-remark-images plugin should be sandwiched between gatsby-remark-extract-image-attributes and gatsby-remark-images-insert-wrapper-attributes, like so:  

``` 
// gatsby-config.js
{
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          `gatsby-remark-extract-image-attributes`,
          `gatsby-remark-images`,
          `gatsby-remark-images-insert-wrapper-attributes`
        ]
      }
}
```

## Front-End Implementation
This plugin is (ultimately) intended to be used with the [CSS attr() selector](https://developer.mozilla.org/en-US/docs/Web/CSS/attr()) to size and align images based on the wrapper dataset attributes that are set:

```
// generated wrapper
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px; width: 300px; height: 200px" data-align="left" data-width="300" data-height="200"> # image source sets</span>

// CSS attr() selectors -- NOT YET FULLY SUPPORTED BY ALL BROWSERS
  [data-width] {
      width: attr(data-width px);
  }

  [data-height] {
      height: attr(data-height px);
  }

  [data-align] {
      float: attr(data-align);
  }

```

However, since attr() isn't fully implemented yet (as of May 31st, 2021), @media rules can be used to unset wrapper CSS styles to achieve the same result:

```
// generated wrapper
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px; width: 300px; height: 200px" data-align="left" data-width="300" data-height="200"> # image source sets</span>

// CSS classes to align image on screens larger than 640px

@media only screen and (min-width:641px) {

  [data-align="right"] {
      float: right;
      margin: 6px 0 10px 40px !important;
  }

  [data-align="left"] {
      float: left;
      margin: 6px 40px 10px 0 !important;
  }

  figure[data-align] {
      margin-bottom: 60px !important;
  }

// CSS classes to remove width and height on screens below 640px in width

@media only screen and (max-width:640px) {

  .gatsby-resp-image-wrapper {
      width: unset !important;
      height: unset !important;
  }

  .gatsby-resp-image-figure {
      width: unset !important;
      height: unset !important;
  }

}

```

If you're using URL params, you can use the following CSS to set (for example) the alignment ([from css-tricks.com](https://css-tricks.com/almanac/selectors/a/attribute/)):

```
img[src*='align=left'] {
  float: left;
  margin: 6px 40px 10px 0;
}

img[src*='align=center'] {
  display: block;
  margin: 0 auto;
}

img[src*='align=right'] {
  float: right;
  margin: 6px 0 10px 40px;
}

```

## Options 

### `setCssInWrapper` (default: true)
You can disable the setting of CSS styles in the wrapper by setting setCSSInWrapper to false. Data attributes will still be set:

``` 
// gatsby-config.js
{
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          `gatsby-remark-extract-image-attributes`,
          `gatsby-remark-images`,
          {
            resolve: `gatsby-remark-images-insert-wrapper-attributes`,
            options: {
              setCssInWrapper: false
            }
          },
        ]
      }
}

// resulting wrapper styles and attributes
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px" data-width="300" data-height="200" data-align="left"> # image source sets</span>

```
### `alignedImageWidth` (default: 300)
You can use the "alignedImageWidth" option to set a maximum width for images that are left- or right-aligned:

``` 
// gatsby-config.js
{
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          `gatsby-remark-extract-image-attributes`,
          `gatsby-remark-images`,
          {
            resolve: `gatsby-remark-images-insert-wrapper-attributes`,
            options: {
              alignedImageWidth: 300
            }
          },
        ]
      }
}

```
 The plugin will calculate the correct pixel dimensions and set the CSS width and height accordingly. If this option is set, it can only be overridden by a smaller width setting in the markdownAST property for that image.

*Example 1:*
- "alignedImageWidth" is set to 300
- The width and height that are set in the wrapper will be set to height: 300px, width: 200px

```
// markdownAST
{
    type: 'root',
    children: [
      {
        type: 'html',
        value: image HTML data
        position: [positionData]
        align: 'left',
        width: '600',
        height: '400',
        caption: null
        },
    ],
    position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 93, column: 1, offset: 21282 }
    }

// resulting wrapper styles and attributes
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px; width: 300px; height: 200px" data-width="300" data-height="200" data-align="left"> # image source sets</span>

```

*Example 2:*
- "alignedImageWidth" is set to 300
- Because the attributes in the AST are below 300, the width and height that are set in the wrapper will be set to height: 150px, width: 100px

```
// markdownAST
{
    type: 'root',
    children: [
      {
        type: 'html',
        value: image HTML data
        position: [positionData]
        align: 'left',
        width: '150',
        height: '100',
        caption: null
        },
    ],
    position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 93, column: 1, offset: 21282 }
    }

// resulting wrapper styles and attributes
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px; width: 150px; height: 100px" data-width="150" data-height="100" data-align="left"> # image source sets</span>

```
*Example 3:*
- "alignedImageWidth" is set to 1200
- The width and height that are set in the wrapper will be set to height: 600px, width: 400px

```
// markdownAST
{
    type: 'root',
    children: [
      {
        type: 'html',
        value: image HTML data
        position: [positionData]
        align: 'left',
        width: '600',
        height: '400',
        caption: null
        },
    ],
    position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 93, column: 1, offset: 21282 }
    }

// resulting wrapper styles and attributes
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px; width: 600px; height: 400px" data-width="600" data-height="400" data-align="left"> # image source sets</span>

```
*Example 4:*
- "alignedImageWidth" is set to 300
- Width and height attributes are NOT set in the AST
- The system will calculate width and height based on the dimensions of the image file itself
- Let's say that the image file is 600x400 pixels
- The styles that are set in the wrapper will be set to height: 300px, width: 200px

```
// markdownAST
{
    type: 'root',
    children: [
      {
        type: 'html',
        value: image HTML data
        position: [positionData]
        align: 'left'
        caption: null
        },
    ],
    position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 93, column: 1, offset: 21282 }
    }

// resulting wrapper styles and attributes
<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 930px; width: 300px; height: 200px" data-width="300" data-height="200" data-align="left"> # image source sets</span>

```

## Testing

To run Jest tests: 
```
npm test
```

changelog.md
# Changelog
All notable changes to this project will be documented in this file.

## [1.0.3] - 2021-06-02
### Changed
- Updated example 4 of README file
- Added changelog.md

## [1.0.2] - 2021-06-01
This should be considered the initial release

### Changed
- Fixed main source in package.json






src/__tests__/index.test.js
const plugin = require('../index');
const {mockAst, filesMock} = require('../__mocks__/ast.mocks');
const jsdom = require("jsdom");
const { JSDOM } = jsdom;

describe("AST visit test", () => {

    it("should insert attributes into wrapper", done => {
        const pluginOptions = {setCssInWrapper: true, alignedImageWidth: 300};
        const files = filesMock;
        files[0].absolutePath = __dirname + "/images/test.jpg";
        const ast = {markdownAST: mockAst, files: files};
        const newAst = plugin(ast, pluginOptions);
        const {type, value, position, ...attributes} = newAst.children[0];
        const dom = new JSDOM(value, { contentType: "text/html" });
        const imgWrapper = dom.window.document.getElementsByClassName("gatsby-resp-image-wrapper")[0];
        expect(imgWrapper.style.width).toBe("300px");
        expect(imgWrapper.style.height).toBe("200px");
        expect(imgWrapper.dataset.width).toBe("300");
        expect(imgWrapper.dataset.height).toBe("200");
        expect(imgWrapper.dataset.align).toBe("left");
        done();
    })

})

src/__mocks__/ast.mocks.js
const nodeMock = {
    type: 'html',
    value: `<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 720px;
  ">
    <a class="gatsby-resp-image-link" href="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" style="display: block"
  target="_blank" rel="noopener">
      <span class="gatsby-resp-image-background-image" style="padding-bottom: 100%; position: relative; bottom: 0; left: 0; background-image:
  url(&apos;data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQ
  jhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAXAQEBAQEAAAAAAAA
  AAAAAAAACAQAD/9oADAMBAAIQAxAAAAG7DNiNvMoenSoMVxL/AP/EABsQAAIDAQEBAAAAAAAAAAAAAAECAAMREgQT/9oACAEBAAEFAu881T4itoLM0DQW5LEHzYcsSZ//xAAXEQADAQAAAAAAAAAAA
  AAAAAAAARAR/9oACAEDAQE/AUbf/8QAFhEAAwAAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/ASP/xAAYEAEBAAMAAAAAAAAAAAAAAAABECAxYf/aAAgBAQAGPwI7dYIT/8QAGxAAAwADAQEAAAAAAAA
  AAAAAAAERITFRQXH/2gAIAQEAAT8hcC5wFLXayG2PfGbEPVzfg1IapR9QrQKVbP/aAAwDAQACAAMAAAAQLy9C/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QtOkC7n//xAAZEQEAA
  gMAAAAAAAAAAAAAAAABABARITH/2gAIAQIBAT8QRTcyjyv/xAAcEAEAAwACAwAAAAAAAAAAAAABABEhMWFxofH/2gAIAQEAAT8QLONg9y3ItDe41KF5JnQlWnKq/kYQAWDo2DQX5ZRqIrYiNkH2hAm
  GJ//Z&apos;); background-size: cover; display: block;"></span>
      <picture>
      <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/c85cb/test.webp 300w,
      /static/568c5d417f7ef32ba3bb5b95bdf0f42d/e88ff/test.webp 600w,
      /static/568c5d417f7ef32ba3bb5b95bdf0f42d/bc514/test.webp 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/webp">
     <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/f93b5/test.jpg 300w,
      /static/568c5d417f7ef32ba3bb5b95bdf0f42d/b4294/test.jpg 600w,
      /static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/jpeg">
      <img class="gatsby-resp-image-image" src="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" alt="test"
  title="test" loading="lazy" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;">
      </picture>
      </a>
      </span>`,
    position: {
      start: { line: 68, column: 1, offset: 15708 },
      end: { line: 68, column: 124, offset: 15831 },
      indent: []
    },
    align: 'left',
    width: '600',
    height: '400',
    caption: null
  }

  const nodeCaptionMock = {
    type: 'html',
    value: `<figure class="gatsby-resp-image-figure" style="width: 379px; height: 293px;" data-align="left" data-width="379" data-height="293" data-caption="Deb and her Dad">
    <span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 720px;">
    <a class="gatsby-resp-image-link" href="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" style="display: block" target="_blank" rel="noopener">
    <span class="gatsby-resp-image-background-image" style="padding-bottom: 100%; position: relative; bottom: 0; left: 0; background-image:
    url(&apos;data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQ
    jhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAXAQEBAQEAAAAAAAA
    AAAAAAAACAQAD/9oADAMBAAIQAxAAAAG7DNiNvMoenSoMVxL/AP/EABsQAAIDAQEBAAAAAAAAAAAAAAECAAMREgQT/9oACAEBAAEFAu881T4itoLM0DQW5LEHzYcsSZ//xAAXEQADAQAAAAAAAAAAA
    AAAAAAAARAR/9oACAEDAQE/AUbf/8QAFhEAAwAAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/ASP/xAAYEAEBAAMAAAAAAAAAAAAAAAABECAxYf/aAAgBAQAGPwI7dYIT/8QAGxAAAwADAQEAAAAAAAA
    AAAAAAAERITFRQXH/2gAIAQEAAT8hcC5wFLXayG2PfGbEPVzfg1IapR9QrQKVbP/aAAwDAQACAAMAAAAQLy9C/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QtOkC7n//xAAZEQEAA
    gMAAAAAAAAAAAAAAAABABARITH/2gAIAQIBAT8QRTcyjyv/xAAcEAEAAwACAwAAAAAAAAAAAAABABEhMWFxofH/2gAIAQEAAT8QLONg9y3ItDe41KF5JnQlWnKq/kYQAWDo2DQX5ZRqIrYiNkH2hAm
    GJ//Z&apos;); background-size: cover; display: block;"></span>
    <picture>
    <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/c85cb/test.webp 300w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/e88ff/test.webp 600w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/bc514/test.webp 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/webp">
    <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/f93b5/test.jpg 300w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/b4294/test.jpg 600w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/jpeg">
    <img class="gatsby-resp-image-image" src="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" alt="test"
    title="test" loading="lazy" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;">
    </picture>
    </a>
    </span>
    <figcaption class="gatsby-resp-image-figcaption"><p>Deb and her Dad</p></figcaption>
    </figure>`,
    position: {
        start: { line: 68, column: 1, offset: 15708 },
        end: { line: 68, column: 124, offset: 15831 },
        indent: []
    },
    align: 'left',
    width: '600',
    height: '400',
    caption: null
}

  const astMock = {
    type: 'root',
    children: [
      {
        type: 'html',
        value: `<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 720px;">
        <a class="gatsby-resp-image-link" href="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" style="display: block" target="_blank" rel="noopener">
        <span class="gatsby-resp-image-background-image" style="padding-bottom: 100%; position: relative; bottom: 0; left: 0; background-image:
        url(&apos;data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQ
        jhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAXAQEBAQEAAAAAAAA
        AAAAAAAACAQAD/9oADAMBAAIQAxAAAAG7DNiNvMoenSoMVxL/AP/EABsQAAIDAQEBAAAAAAAAAAAAAAECAAMREgQT/9oACAEBAAEFAu881T4itoLM0DQW5LEHzYcsSZ//xAAXEQADAQAAAAAAAAAAA
        AAAAAAAARAR/9oACAEDAQE/AUbf/8QAFhEAAwAAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/ASP/xAAYEAEBAAMAAAAAAAAAAAAAAAABECAxYf/aAAgBAQAGPwI7dYIT/8QAGxAAAwADAQEAAAAAAAA
        AAAAAAAERITFRQXH/2gAIAQEAAT8hcC5wFLXayG2PfGbEPVzfg1IapR9QrQKVbP/aAAwDAQACAAMAAAAQLy9C/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QtOkC7n//xAAZEQEAA
        gMAAAAAAAAAAAAAAAABABARITH/2gAIAQIBAT8QRTcyjyv/xAAcEAEAAwACAwAAAAAAAAAAAAABABEhMWFxofH/2gAIAQEAAT8QLONg9y3ItDe41KF5JnQlWnKq/kYQAWDo2DQX5ZRqIrYiNkH2hAm
        GJ//Z&apos;); background-size: cover; display: block;"></span>
        <picture>
        <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/c85cb/test.webp 300w,
        /static/568c5d417f7ef32ba3bb5b95bdf0f42d/e88ff/test.webp 600w,
        /static/568c5d417f7ef32ba3bb5b95bdf0f42d/bc514/test.webp 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/webp">
        <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/f93b5/test.jpg 300w,
        /static/568c5d417f7ef32ba3bb5b95bdf0f42d/b4294/test.jpg 600w,
        /static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/jpeg">
        <img class="gatsby-resp-image-image" src="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" alt="test"
        title="test" loading="lazy" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;">
        </picture>
        </a>
        </span>`,
        position: {
              start: { line: 68, column: 1, offset: 15708 },
              end: { line: 68, column: 124, offset: 15831 },
              indent: []
        },
        align: 'left',
        width: '526',
        height: '526',
        caption: null
        },
    ],
    position: {
        start: { line: 1, column: 1, offset: 0 },
        end: { line: 93, column: 1, offset: 21282 }
    }
}

const nodeImageAttributesMock = {
  type: 'html',
  value: `<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 720px;
">
  <a class="gatsby-resp-image-link" href="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" style="display: block"
target="_blank" rel="noopener">
    <span class="gatsby-resp-image-background-image" style="padding-bottom: 100%; position: relative; bottom: 0; left: 0; background-image:
url(&apos;data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQ
jhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAXAQEBAQEAAAAAAAA
AAAAAAAACAQAD/9oADAMBAAIQAxAAAAG7DNiNvMoenSoMVxL/AP/EABsQAAIDAQEBAAAAAAAAAAAAAAECAAMREgQT/9oACAEBAAEFAu881T4itoLM0DQW5LEHzYcsSZ//xAAXEQADAQAAAAAAAAAAA
AAAAAAAARAR/9oACAEDAQE/AUbf/8QAFhEAAwAAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/ASP/xAAYEAEBAAMAAAAAAAAAAAAAAAABECAxYf/aAAgBAQAGPwI7dYIT/8QAGxAAAwADAQEAAAAAAAA
AAAAAAAERITFRQXH/2gAIAQEAAT8hcC5wFLXayG2PfGbEPVzfg1IapR9QrQKVbP/aAAwDAQACAAMAAAAQLy9C/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QtOkC7n//xAAZEQEAA
gMAAAAAAAAAAAAAAAABABARITH/2gAIAQIBAT8QRTcyjyv/xAAcEAEAAwACAwAAAAAAAAAAAAABABEhMWFxofH/2gAIAQEAAT8QLONg9y3ItDe41KF5JnQlWnKq/kYQAWDo2DQX5ZRqIrYiNkH2hAm
GJ//Z&apos;); background-size: cover; display: block;"></span>
    <picture>
    <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/c85cb/test.webp 300w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/e88ff/test.webp 600w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/bc514/test.webp 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/webp">
   <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/f93b5/test.jpg 300w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/b4294/test.jpg 600w,
    /static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/jpeg">
    <img class="gatsby-resp-image-image" src="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" alt="test"
title="test" loading="lazy" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;">
    </picture>
    </a>
    </span>`,
  position: {
    start: { line: 68, column: 1, offset: 15708 },
    end: { line: 68, column: 124, offset: 15831 },
    indent: []
  },
  align: 'left',
  caption: null
}

const filesMock = [{
  id: '725468ea-90ea-505c-bbf3-af86b9029ec0',
  children: [ 'ecbec921-e3b4-5386-9cdc-fc7962ded21f' ],
  parent: null,
  internal: {
    contentDigest: '9cc3f55fcb4dcdf9a056d51498d9415f',
    type: 'File',
    mediaType: 'image/jpeg',
    description: 'File "/images/test.jpg"',
    owner: 'gatsby-source-filesystem',
    counter: 1570
  },
  sourceInstanceName: 'pages',
  relativePath: 'images/test.jpg',
  extension: 'jpg',
  prettySize: '97.5 kB',
  modifiedTime: '2021-03-04T21:36:49.058Z',
  accessTime: '2021-03-04T21:37:07.229Z',
  changeTime: '2021-03-04T21:37:01.791Z',
  birthTime: '2021-03-04T21:36:49.056Z',
  root: '/',
  dir: `/images`,
  base: 'test.jpg',
  ext: '.jpg',
  name: 'test',
  absolutePath: `/images/test.jpg`,
  relativeDirectory: 'images',
  dev: 16777220,
  mode: 33188,
  nlink: 1,
  uid: 501,
  rdev: 0,
  blksize: 4096,
  ino: 594269138,
  size: 97518,
  blocks: 192,
  atimeMs: 1614893827228.6414,
  mtimeMs: 1614893809058.3716,
  ctimeMs: 1614893821790.7444,
  birthtimeMs: 1614893809055.759,
  atime: '2021-03-04T21:37:07.229Z',
  mtime: '2021-03-04T21:36:49.058Z',
  ctime: '2021-03-04T21:37:01.791Z',
  birthtime: '2021-03-04T21:36:49.056Z',
  __gatsby_resolved: undefined
}]

const mockAst = {
  "type": "root",
  "children": [
    {
      "type": "html",
      "value": `<span class="gatsby-resp-image-wrapper" style="position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 720px;">
      <a class="gatsby-resp-image-link" href="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" style="display: block" target="_blank" rel="noopener">
      <span class="gatsby-resp-image-background-image" style="padding-bottom: 100%; position: relative; bottom: 0; left: 0; background-image:
      url(&apos;data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQ
      jhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAXAQEBAQEAAAAAAAA
      AAAAAAAACAQAD/9oADAMBAAIQAxAAAAG7DNiNvMoenSoMVxL/AP/EABsQAAIDAQEBAAAAAAAAAAAAAAECAAMREgQT/9oACAEBAAEFAu881T4itoLM0DQW5LEHzYcsSZ//xAAXEQADAQAAAAAAAAAAA
      AAAAAAAARAR/9oACAEDAQE/AUbf/8QAFhEAAwAAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/ASP/xAAYEAEBAAMAAAAAAAAAAAAAAAABECAxYf/aAAgBAQAGPwI7dYIT/8QAGxAAAwADAQEAAAAAAAA
      AAAAAAAERITFRQXH/2gAIAQEAAT8hcC5wFLXayG2PfGbEPVzfg1IapR9QrQKVbP/aAAwDAQACAAMAAAAQLy9C/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QtOkC7n//xAAZEQEAA
      gMAAAAAAAAAAAAAAAABABARITH/2gAIAQIBAT8QRTcyjyv/xAAcEAEAAwACAwAAAAAAAAAAAAABABEhMWFxofH/2gAIAQEAAT8QLONg9y3ItDe41KF5JnQlWnKq/kYQAWDo2DQX5ZRqIrYiNkH2hAm
      GJ//Z&apos;); background-size: cover; display: block;"></span>
          <picture>
          <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/c85cb/test.webp 300w,
          /static/568c5d417f7ef32ba3bb5b95bdf0f42d/e88ff/test.webp 600w,
          /static/568c5d417f7ef32ba3bb5b95bdf0f42d/bc514/test.webp 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/webp">
         <source srcset="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/f93b5/test.jpg 300w,
          /static/568c5d417f7ef32ba3bb5b95bdf0f42d/b4294/test.jpg 600w,
          /static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg 720w" sizes="(max-width: 720px) 100vw, 720px" type="image/jpeg">
          <img class="gatsby-resp-image-image" src="/static/568c5d417f7ef32ba3bb5b95bdf0f42d/80e3c/test.jpg" alt="test"
      title="test" loading="lazy" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;">
          </picture>
          </a>
          </span>`,
      "width": "300",
      "height": "200",
      "align": "left",
      "position": {
        "start": {
          "line": 1,
          "column": 1,
          "offset": 0
        },
        "end": {
          "line": 1,
          "column": 95,
          "offset": 94
        }
      },
    }
  ],
  "position": {
    "start": {
      "line": 1,
      "column": 1,
      "offset": 0
    },
    "end": {
      "line": 1,
      "column": 95,
      "offset": 94
    }
  }
}

module.exports = {nodeMock, nodeCaptionMock, astMock, nodeImageAttributesMock, filesMock, mockAst};